core/src/main/scala/org/zalando/grafter/Rewriter.scala

```diff
@@ -218,3 +239,8 @@ trait RewriterSyntax {
 
 object RewriterSyntax extends RewriterSyntax
 
+private object GrafterMemoRewriter extends MemoRewriter {
+  override def dup[T <: Product](t : T, children : Array[AnyRef]) : T =
+    try super.dup(t, children)
```

Comment:
We don't fail hard on rewriting for case classes like `org.http4s.Uri` which fail to be rewritten (maybe because that class is final, but otherwise it can be because that class is nested).